pkg/asset/agent/installconfig_test.go

```diff
@@ -40,26 +45,53 @@ pullSecret: "{\"auths\":{\"example.com\":{\"auth\":\"authorization value\"}}}"
 			expectedError: `invalid install-config configuration: Platform: Unsupported value: "aws": supported values: "baremetal", "vsphere", "none"`,
 		},
 		{
-			name: "apiVips not set for baremetal platform",
+			name: "apiVips not set for baremetal Compact platform",
 			data: `
 apiVersion: v1
 metadata:
   name: test-cluster
 baseDomain: test-domain
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14 
+    hostPrefix: 23 
+  networkType: OpenShiftSDN
+  machineNetwork:
+  - cidr: 192.168.122.0/23
+  serviceNetwork: 
+  - 172.30.0.0/16
+compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
 platform:
   baremetal:
+    externalMACAddress: "52:54:00:f6:b4:02"
+    provisioningMACAddress: "52:54:00:6e:3b:02"
+    ingressVIPs: 
+      - 192.168.122.11
     hosts:
       - name: host1
-        bootMACAddress: 52:54:01:xx:zz:z1
-    ingressVips:
-      - 192.168.122.11
+        bootMACAddress: 52:54:01:aa:aa:a1
+      - name: host2
+        bootMACAddress: 52:54:01:bb:bb:b1
+      - name: host3
+        bootMACAddress: 52:54:01:cc:cc:c1
 pullSecret: "{\"auths\":{\"example.com\":{\"auth\":\"authorization value\"}}}"
 `,
 			expectedFound: false,
-			expectedError: "invalid install-config configuration: Platform.Baremetal.ApiVips: Required value: apiVips must be set for baremetal platform",
+			expectedError: "failed to create install config: invalid \"install-config.yaml\" file: [platform.baremetal.apiVIPs: Invalid value: []string{\"DNS lookup failure: lookup api.test-cluster.test-domain on 10.0.80.11:53: no such host\"}: ip <nil> is invalid, platform.baremetal.apiVIPs: Invalid value: \"DNS lookup failure: lookup api.test-cluster.test-domain on 10.0.80.11:53: no such host\": \"DNS lookup failure: lookup api.test-cluster.test-domain on 10.0.80.11:53: no such host\" is not a valid IP, platform.baremetal.apiVIPs: Invalid value: \"DNS lookup failure: lookup api.test-cluster.test-domain on 10.0.80.11:53: no such host\": IP expected to be in one of the machine networks: 192.168.122.0/23]",
```

Comment:
Now that we first do the validation by the default install config, we get these wordy messages

Reply:
This is failing in CI because the DNS IP (10.0.80.11:53 here) depends on your local network.
Also this error message is pretty terrible, somebody should really clean it up in the main InstallConfig code.